Removes bolt of death from xenobiology magicarps.

AcapitalA opened the pull request:
The bolt of death has been removed from the xenobiology Magicarp and Chaos Magicarp.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Removes bolt of death on the xenobiology Magicarp and Chaos Magicarp. Can't be more to-the-point than that.

I've decided to keep the other bolts in for magicarps, because those can be fun in the right hands, but bolts of death are simply unfun and unbalanced no matter what, often on both ends.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The bolt of death is what it says on the tin: A bolt of death.

We removed tasers, SRM-8 missile racks, nerfed rubbershot and tried to get rid of so many other oneshots for a reason: They're unfun to play against and count as an instant win in combat. It would also be one less reason for powergamers to go into xenobiology.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
balance: In a freak magic accident, all xenobio-made magicarps have lost their bolt of death. This is a great tragedy for xenobiologists all around the galaxy.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

AcapitalA committed:
Removes bolts of death from all magicarps.

The bolt of death has been removed from the Magicarp and Chaos Magicarp.

Timonkeyn commented:
ided

Ty-the-Smonk commented:
as a xenobio autist i approve of this pr

Timonkeyn commented:
just get antimagic bro

Timonkeyn commented:
bolt of death magicarps are rare anyways, but you made this PR because there was one in the round before the current round on terry

Timonkeyn commented:
apart from that, the ones you changed Arent even the xenobio carps, its the magicarp wizard Event that you changed

AcapitalA commented:
> bolt of death magicarps are rare anyways, but you made this PR because there was one in the round before the current round on terry

As I said before, we removed instant wins in combat for several reasons.

Tasers got removed because they were an instant win.
SRM-8 missile racks got removed because they were an instant win.
The Scattershot got nerfed because up close it was an instant win.
Rubbershot got nerfed because up close it was an instant win.
Seperated chems got removed because it allowed for instant win bombs.
Explosive lances got nerfed because they were instant win bombs.
Flashbangs got nerfed because they could be instant win grenades.

If the community can come to a conclusion, It's that instant wins are unfair and unbalanced.

Also, magicarp files are supposed to be in there. Tested it on localhost and it works 100%

Time-Green commented:
> apart from that, the ones you changed Arent even the xenobio carps, its the magicarp wizard Event that you changed

The file they are in doesn't necessarily decide what object it is, that is a strangely common misconception. Xenobiology carp are wizard carp and the change works.

Angustmeta commented:
Getting bolt of death on two quick-spawned magicarps was a happy surprise. 

I don't mind this PR but at least people killed by it are easily defibbable, unlike when killed by fireballs and such.

AcapitalA commented:
> Getting bolt of death on two quick-spawned magicarps was a happy surprise.
> I don't mind this PR but at least people killed by it are easily defibbable, unlike when killed by fireballs and such.

Yeah, saw it happen after you managed to gib half the server with the new traitor item. That round was more of a reminder than an actual reason for it however.
I knew how powerful deathbolt magicarps are because instant death isn't exactly something you poke at with a stick, but didn't bother back then because I couldn't code with BYOND yet.
Honestly suprised with the new traitor item bone you did, was pretty damn creative if anything.

Angustmeta commented:
A bit OOT but I didn't use any traitor item to gib half the server. It was just a chilling bluespace crossbreed. Which definitely needs a change to the size/color of the alert.

AcapitalA commented:
> A bit OOT (still xenobio related) but I didn't use any traitor item to gib half the server. It was just a chilling bluespace crossbreed. Which definitely needs a change to the size/color of the alert.

Thought the traitor implant place swapper was merged, suprised you managed to pull it off with sepias. If anything, that's even more impressive.

Timonkeyn commented:
> > apart from that, the ones you changed Arent even the xenobio carps, its the magicarp wizard Event that you changed
> 
> The file they are in doesn't necessarily decide what object it is, that is a strangely common misconception. Xenobiology carp are wizard carp and the change works.

how do xenobio magicarps get bolt of animation etc. then?

AcapitalA commented:
> how do xenobio magicarps get bolt of animation etc. then?
Because they ARE wizard carp. Xenobiology and wizard event magicarps are the same thing.

nemvar commented:
I firmly remember just randomly dying 20 minutes into the round because some chemist mixed life and a death magicarp shot a bolt (not even aimed at me) down the hallway.

Magic should be contained to the magic gamemodes, magicarps should never have entered the random spawn pool in the first place.

AcapitalA commented:
> I firmly remember just randomly dying 20 minutes into the round because some chemist mixed life and a death magicarp shot a bolt (not even aimed at me) down the hallway.
> Magic should be contained to the magic gamemodes, magicarps should never have entered the random spawn pool in the first place.

Exactly.

Originally I was planning to remove magicarps from xenobio as a whole because they aren't balanced if the gamemode isn't wizard, but considering ~~how much backlash that would get~~ magicarps also have somewhat more balanced / fun bolts I simply decided to remove the most abused and overpowered one.

OverlyGeneric commented:
> Originally I was planning to remove magicarps from xenobio as a whole because they aren't balanced if the gamemode isn't wizard, but considering ~how much backlash that would get~ magicarps also have somewhat more balanced / fun bolts I simply decided to remove the most abused and overpowered one.

do it then. don't be a big pussycat.

ATH1909 commented:
Just use a defib, bro.

Death magicarps (especially SAPIENT ones, since it can be hard to tell what bolt a magicarp has) are super rare, anyway- I don't think that they're common enough to really be a problem.

If you really want to remove deathcarps, remove the death bolt from normal magicarps but keep it in the chaos magicarp, since the chaos magicarp can't reliably fire death bolts.

AcapitalA commented:
> Just use a defib, bro.

Because a defib or cloner should be the only remedy for a single shot from a magicarp.
> Death magicarps (especially SAPIENT ones, since it can be hard to tell what bolt a magicarp has) are super rare, anyway- I don't think that they're common enough to really be a problem.

Something being rare doesn't mean it can't be overpowered / unfair to play against. Imagine if there was a chance of 1% for each antag to randomly get an instakill rifle.
> If you really want to remove deathcarps, remove the death bolt from normal magicarps but keep it in the chaos magicarp, since the chaos magicarp can't reliably fire death bolts.

RNG shouldn't affect combat as much as possible. It throws skill out of the window and it only incites "I would have won if RNG mechanic didn't exist" arguements.

AdamElTablawy commented:
Just make a subtype of magicarp that can be spawned with xeno missing the death bolt and keep regular magicarps, now unspawnable, with death bolts and chaos carp with their deathbolts.

Chaos carp having death bolts is fine as they’re highly random and it adds some risk to shrieking TRANSFORM ME at the damn thing because you might full on die and be teleported, and are also a wizard transform ability. Given that the wizard has what amounts to four separate oneshot abilities his chaos carp TF having deathbolt is fine.

knacker48 commented:
I doubt this will sway anybodies opinion on this but its worth noting that carps have very low health. As in they die in two laser hits which is part of the reason you can't kill the entire station with carps alone (that and the rng and time investment it takes to make em in the first place)

ExcessiveUseOfCobblestone commented:
I'd rather just see the stupid magic fish removed from xenobio pool, I've never understood why we NEEDED wizard spells for slime farmers when they can generate literal armies on a consistent basis (and these aren't even mutually exclusive).

ExcessiveUseOfCobblestone commented:
waiting for the shadowflame post tbh

AcapitalA commented:
> I'd rather just see the stupid magic fish removed from xenobio pool, I've never understood why we NEEDED wizard spells for slime farmers when they can generate literal armies on a consistent basis (and these aren't even mutually exclusive).

Yeah, wizard gear for any round that isn't wizard is unbalanced as hell. As I said before, was planning to remove the carp from xenobio, but considering the general fuckery you could do with it besides murderboning I just went for the death bolts only.

Maybe sometime later for the entire carp, but for now removing the most dangerous part sounds like a good step in the right direction.

Shadowflame909 commented:
I've already made that page long rant @ExcessiveUseOfCobblestone the last time someone tried to remove magicarps from the gold slime pool.

Headslugs have been taken.

The Statue-Simple Mob is now admin only.

Spiders are syndicate only.

I'm really not against this change. Because Slime-Farmers holy grail is Bolt of Change.

Even this itself isn't too outrageous in my opinion though.

You willingly change into a valid mob. You'll get banned for griefing, and your main functions are to help the station or scare the crap out of them.

Its real spooks are 1. The 1 in 300 chance of lesser ash-drake. 2. Syndicate Borg. 3. Xeno.

A toolbox, a flash, and a welder/shot-gun are all the necessity someone needs to get their valids on these mobs. With self-defense only being a tool after the fact.

The only other Mob in the golden slime pool worth a hitch would be the Pine Tree? Because they have a stun chance on hit. Pretty strong, but only antags or griefers would really find a use for them.

Sepia Slimes now have a big down-side. Crossbreeding itself strays away from any "OP" mechanics and instead makes sillier reactions that would be more deadly to yourself first before anyone else. The strongest weapon in it is RNG locked to the end of xenobiology with Rainbow Slimes. So that'd be about the same time you can pull off Carp RNG if you're lucky.

Some people want a game where the only benefit from the departments, is a warm feeling in your heart where you give someone else something and that's all your capable of. I personally think /tg/ has never been that server, and the rules certainly don't reflect that mentality. Otherwise, escalation would be a thing of the past. There's only such much you can try to do with the code to change that.

As Kor once said. Yep, Magicarps are the holy grail of xenobiology. I still think that stands because nothing else can live up to the fun buzz of playing the Species Change lottery.

But Tldr; I don't mind this, because it takes some of the worst RNG away from my Xenobiology end-game. I would be against a full removal. Because I don't see any non neutered value in a Xenobio end-game without it. That is truly, comparable to all the other departments.

But...if things did go the route of full removal. Maybe an icon-changer slime would be fun. I'd still get my kicks from seeing the crew run around in distress.

PKPenguin321 commented:
These are genuinely some of the funniest things in the game, please don't remove

ExcessiveUseOfCobblestone reviewed (changes requested):
The wizard should not be affected by this change.

ExcessiveUseOfCobblestone commented:
Just remove from slime pool tbh I disagree with the idea that I should be setting xenobio's power to how it was before features were changed because they were deemed too strong/negatively impact the game.

Dawson1917 commented:
1-shot kills have always and will always be bad for the game. 
If you need any more reason than that to get rid of death bolt, ask yourself "who finds this enjoyable?"
Non-antag xenobiologists? No use except validhunting and inevitably getting nonantags killed when the fish takes its orders overboard. Or gets bullied.
Antag xenobiologists? They send the fish out to nothing personal everyone in the hallway with their magic delete projectile. The xenobiologist plays a very inactive and boring part, and everyone killed by the fish dies in the least interesting way imaginable.

@ExcessiveUseOfCobblestone imo the other magicarp types have merit. Resurrection is a very powerful and beneficial tool to the crew from xenobio doing their job right, and bolt of change is either for giving the xenobiologists big pp points or turning (usually) willing crew into xenomorphs or syndiborgs. People normally enjoy it.

nemvar commented:
Imagine thinking that having access to a literal bolt of resurrection is a good thing. We are trying to make medical meaningful and certain shucklefucks thing it's alright to have literal on demand aheals in no short supply.
Oh yes, this one magic fish that you can easily powergame in 30 minutes and literally makes death/injuries meaningless is very conductive to engaging gameplay. Totally not like it makes an entire department completely redudant.

What xenobio needs is an identity instead of being the swiss army knife of powergamers that gets everything except guns.

Dawson1917 commented:
> What xenobio needs is an identity instead of being the swiss army knife of powergamers that gets everything except guns.

(gatfruit)

ExcessiveUseOfCobblestone commented:
I could probably be sold on a tele fish and an arcane barrage fish. That's about it.

Shadowflame909 commented:
Cowman must see the value in change carp. It has more downsides for non antags then upsides


You can also turn into a cow with it

AcapitalA commented:
Magicarps in general shouldn't be in xenobiology, since it's meant to be a wizard event mob, and wizards are literally the entire crew VS a single person with god-tier powers.

As I said before: I will only remove the bolt of death in this PR for several reasons, but then again I also see why the magicarp should be removed from xenobiology as a whole. This PR just seems as a good middle-ground for the entire magicarp debate.

Shadowflame909 commented:
"Xenobio is OP" is the same level of meme as "Clockcult is bad and there's nothing you can do about it."


Xenobio endgame is what? Magicarp of Change? Sepia Grenades?

It's pretty much just an industrial complex with slimes otherwise. A monkey maker, metal maker, not much actual interesting content to experiment with outside of the main carp IMO.

More and more things just keep getting removed. But xenobio is pretty much the only job in science. Bomb making is grief or used for RND and RND isn't a job.

To further my point of how weaksauce xenobio is without the MoC.

MD endgame: Romerol zombies and pacification, and ling reviving stasis lite. Also skewium because chemistry has nothing better to do then grief. Sad.

Mining: Rarely, lesser ash Drake. More commonly, antag, immortality or a stand.

Cargo: Guns.

Robotics: Giant armor and guns as well.

Assistant: Nukie hardsuit, Revolver.

Come on gamers.

ExcessiveUseOfCobblestone commented:
SoC for xenobio is dumb because the "SoC" IS the gold slime.

You're literally bypassing the balance regarding what is meant to be present as part of the gold core arsenal. If you want mobs spawnable by xeno then toss them into the gold slime pool directly. If you can't have them in the gold slime pool then wow they probably shouldn't have been there to begin with by just having a magic fish.

Not going to bother entertaining things like guns and hulk (lol'd) are somehow comparable to wizard spells. If you think some of the other items are incredibly strong (which is fair) then review those instead of arms racing with them, that's literally textbook powercreep.

Be sure to add the cow to the pool btw.

Shadowflame909 commented:
The effort reflects the content. This has always been the case with every  job in this game. With the most fun things requiring the highest level of effort. Like Romerol via medbay, a phazon, cake Kat. This is even reflected in the chaplains chance of getting a response from an admin.

This is the peak of xenobio. You'd be bleaving a gap that would be detrimental to the game in the form of less content people want to work towards. As seen by the current state of science.

The same reason Kor denied the PR previously because even if lesser. It needs some sort of magnum opus.

At the very least. I'd request that you deny any pure removal PR that doesn't attempt to replace the content with something more healthy to end-game balance to strive for.

Science is in a slump. But you wouldn't remove magicarps for the same reason you wouldn't remove rewards for killing megafauna or tendrils on a lesser scale.

You're not giving them a lengthy engaging goal to work towards.

PS. This items RNG is comparable to rolling a die of fate. By the time you get it. The round is already over. End game content is most definitely harder to achieve and that much more powerful then what you start with. Which is just stating the obvious.

ExcessiveUseOfCobblestone commented:
Then **you** add more content to science that is balanced rather than demand I pigeonhole balance of an entire department at the cost of the rest of the game because "they have nothing else". Whatever Kor decided was based on the current iteration of the game, I'm basing my decision off the same metric. If he said something equivalent to xenobio needs infinite ammo ranged wizard projectiles regardless of the rest of the game because their current content is lacking, he was wrong.

Seriously man, if you spent half the time it took you to try to defend a cruddy feature you could have easily coded legitimately good mechanics to replace them.

Tlaltecuhtli commented:
magicarps always felt out of place in the game as if i heard "changecarp in escape" i would just throw away whatever i had as it would be pretty much worthless compared to being a xeno/syndiborg/ashdrake

Shadowflame909 commented:
Yeah Yeah. Arguing with the supreme overlord is futile. I'll whip up something.

It's not a very good play for every void of removed content though. As you're experiencing more turmoil with your current patch piece changes to medbay.

Mend wounds was good when it eventually came though.

AcapitalA committed:
Only xenobiology carps are affected.

Wizard event carps still got their bolts of death.

AcapitalA commented:
God, that took some time to fix. Pretty sure there has to be some way to reduce the lines of code on that, but i can't seem to figure it out after 3 hours of codediving and testing over and over again.

Arathianen commented:
@ExcessiveUseOfCobblestone why are magicarps excessively powerful though without bolt of death? Bolt is obviously OP, but transformation is a tiny chance to get something that turns you into a simplemob but is strong. That isn't OP.

AcapitalA committed:
Gives xenobiology magicarps the correct colors.

Fixes a would-be bug where xenobiology magicarps would have normal carp colors. - I really shouldn't have made a commit at 2AM while being tired as hell...

AcapitalA committed:
Removes call with no recieving end. (probably)

Removed a ..() which led to nowhere.

AcapitalA committed:
Adds empty space, removes a tab.

Removes a tab at 82. - Oops.

AcapitalA committed:
Removes another call going nowhere.

Removed another ..(), but for the chaos variant this time. - Travis please I'm getting worried.

AcapitalA commented:
> The wizard should not be affected by this change.

Only xenobiology carps are affected now, took me more time than I expected.

ExcessiveUseOfCobblestone commented:
> @ExcessiveUseOfCobblestone why are magicarps excessively powerful though without bolt of death? Bolt is obviously OP, but transformation is a tiny chance to get something that turns you into a simplemob but is strong. That isn't OP.

From above
> SoC for xenobio is dumb because the "SoC" IS the gold slime.
> 
> You're literally bypassing the balance regarding what is meant to be present as part of the gold core arsenal. If you want mobs spawnable by xeno then toss them into the gold slime pool directly. If you can't have them in the gold slime pool then wow they probably shouldn't have been there to begin with by just having a magic fish.

Shadowflame909 commented:
I'd enjoy your argument more if everytime a "peak" mob existed it didn't literally get gouged out and removed

>Statues

>Headslugs

I don't think there's any way to add something that people can say. "yes this is my holy-grail of cooperation and research after 1 hour" without it getting removed again

Circular logic cobblestone

ExcessiveUseOfCobblestone commented:
It's almost like they didn't need godmode mobs and changeling either 🤔 

> yes this is my holy-grail of cooperation and research after 1 hour

You can literally make a sentient army in xenobio
You can literally make a sentient army in xenobio
You can literally make a sentient army in xenobio
You can literally make a sentient army in xenobio
You can literally make a sentient army in xenobio
You can literally make a sentient army in xenobio

ExcessiveUseOfCobblestone reviewed (changes requested):
There's no reason to copypasta everything, just subtype from the ranged type and change `allowed_projectile_types` and `gold_core_spawnable`

Shadowflame909 commented:
>thinking that's actually anything worth value

Spiders were the most dangerous thing and now they're traitor only.

The culmination of your non-antag army would be a hive-bot that shoots a 5 brute damage bullet on a cooldown.

A pine tree, a bear and a shark.

ANYONE can come in and bust these weak mobs down in 2 hits.

This isn't a holy-grail. You know this. I know this. Even steve from accounting knows this.

When did it stop being about what made the game fun after 1 hour of work, and instead started turning into "Security has to be the strongest department no matter how much time passes"

It's still Power-Creep. But you're just favoring the most boring department.

ExcessiveUseOfCobblestone commented:
You can still faceroll with numbers on top of the other goodies available to xenobio. They're not in a vacuum. 

Look, I get it you disagree, I'm not going to continue to rehash the same argument over and over.

I will say the target round duration is 2 hours so the hour mark argument is prob not good.

Shadowflame909 commented:
Man.. It's either all vanity and cosmetic items. Or stuff that'll get you banned for using as a non-antag.

Even the second option isn't stackable at all.

>IE using a orange core to start a fire around your gold slime mobs. Same with the oil core to blow stuff up. Same with the sepia slime which freezes them as well.

This was the different one... It's just not worth it.

ExcessiveUseOfCobblestone commented:
You are welcomed to ratify that in a unique way that isn't extremely powerful combat gear equivalent to the power levels of someone who has to take on literally the entire station.

AcapitalA commented:
> There's no reason to copypasta everything, just subtype from the ranged type and change allowed_projectile_types and gold_core_spawnable

I tried before, since it seemed like the most simple and efficient solution, but there's a parent call in there that I can't seem to get rid of / change without breaking the mob's firing ability.

It would call for the normal magicarp projectile list whenever the new list would get called, so I had to make it different sadly.

ExcessiveUseOfCobblestone commented:
you shouldn't be making a new list, you should be overwriting the current one.

aka `xeno_proj_types` should be `allowed_proj_types` for the subtype

AcapitalA commented:
> you shouldn't be making a new list, you should be overwriting the current one.
> aka xeno_proj_types should be allowed_proj_types for the subtype

Didn't even knew how you'd redefine a var, turns out it's actually very simple. I'll make another commit.

AcapitalA committed:
Cleans up code a little.

Compressed code a bit more. - God I love redefines.

ExcessiveUseOfCobblestone commented on code/modules/events/wizard/magicarp.dm:
```diff
@@ -51,7 +52,28 @@
+
+/mob/living/simple_animal/hostile/carp/ranged/chaos/xenobiology/Shoot()
+	projectiletype = pick(allowed_projectile_types)
+	..()
```
67-73 can be removed.

Not only do vars inherit, but so do procs!

AcapitalA replied:
Oh damn, didn't know procs also inherited, this one's gonna save me lots of space. Thanks for that one, I'll get it done in the next commit.

ExcessiveUseOfCobblestone commented on code/modules/events/wizard/magicarp.dm:
```diff
@@ -51,7 +52,28 @@
+	projectiletype = pick(allowed_projectile_types)
+	..()
+
+/mob/living/simple_animal/hostile/carp/ranged/chaos/xenobiology
```
dupe of 61?

AcapitalA replied:
That's the chaos variant, it shoots a random bolt each time instead of a single consistent one.

ExcessiveUseOfCobblestone commented on code/modules/events/wizard/magicarp.dm:
```diff
@@ -36,11 +36,12 @@
-	/obj/item/projectile/magic/spellblade, /obj/item/projectile/magic/arcane_barrage)
-	
+	/obj/item/projectile/magic/teleport, /obj/item/projectile/magic/door, /obj/item/projectile/magic/aoe/fireball, /obj/item/projectile/magic/spellblade,
+	/obj/item/projectile/magic/arcane_barrage, /obj/item/projectile/magic/death)
```
revert? this shouldn't have changed and it'll show up on the tracker as a change.

AcapitalA replied:
Yeah, makes sense considering it basically stayed the same there.

OverlyGeneric commented:
ok, actual opinion time.

remove magicarps.

magic isn't real also let more room available for cool scientific mobs instead of this magical bull crap.

lighgeists are rare as heck, basilisks are only seen in one space ruin, mi-go would be a fun pet for the clown, and a lot of other critters deserve a chance for the gold pool.

just say that magic can't be replicated organically for reasons for the lore.

AcapitalA committed:
Compresses the code even more.

Since procs inherit I managed to save some more space. - Thanks cobb.

ExcessiveUseOfCobblestone commented:
having to make 1line code changes is too daunting, I'd rather write essays.

AcapitalA committed:
Stops tracker from tracking nonexistent changes.

It's the same list as before this PR now.
